
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Fivepseq usage &#8212; fivepseq  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Output interpretation" href="interpreting_output.html" />
    <link rel="prev" title="Preparing Data" href="preparing_data.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="fivepseq-usage">
<span id="usage-fivepseq"></span><h1>Fivepseq usage<a class="headerlink" href="#fivepseq-usage" title="Permalink to this headline">¶</a></h1>
<div class="section" id="command-line-arguments">
<h2>Command line arguments<a class="headerlink" href="#command-line-arguments" title="Permalink to this headline">¶</a></h2>
<p>These are the arguments to be provided:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>fivepseq <span class="se">\</span>
-g &lt;path_to_genome_fasta&gt; <span class="se">\</span>
-a &lt;path_to_annotation&gt; <span class="se">\</span>
-b &lt;path_to_bam_file<span class="o">(</span>s<span class="o">)</span> <span class="se">\ </span> <span class="c1">#(multiple files provided with pattern (&quot;parent_dir/*.bam&quot;))</span>
-o &lt;output_directory&gt; <span class="se">\</span>
-t &lt;title_of_the_run&gt; <span class="se">\</span>
<span class="o">[</span>optional arguments<span class="o">]</span>
</pre></div>
</div>
</div>
<div class="section" id="optional-arguments">
<h2>Optional arguments<a class="headerlink" href="#optional-arguments" title="Permalink to this headline">¶</a></h2>
<p><strong>Conflict</strong></p>
<p>The conflict mode specifies how to deal with files/folders that already exist:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--conflict
    default: add - only adds missing files to existing directory
    overwrite - overwrites all the files in the existing <span class="o">(</span>in <span class="k">case</span><span class="o">)</span> output directory
    alt_dir - uses alternative directory by appending <span class="s1">&#39;+&#39;</span> suffix to existing <span class="o">(</span>in <span class="k">case</span><span class="o">)</span> output directory
</pre></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p><strong>Outlier downsampling</strong></p>
<p>Set p value threshold for outlier detection: counts with less than the [–op] probability of falling into Poisson distribution will be down-sampled.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--op
    default: <span class="m">0</span>
    -1 - turn off downsampling
</pre></div>
</div>
<p>Alternatively, set constant threshold for read count down-sampling. Instead of outlier detection, values larger than this constant will be down-sampled.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>--ds
    default: None
</pre></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p><strong>Gene filter</strong></p>
<p>Supply a text file with newline-separated list of gene ids you’d like to filter/use.
The names should correspond to those present under the gene_id tag in the gff file.
Note: only these genes will be used in all the calculations.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-gf/-genefilter
    default: None
</pre></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<p><strong>Gene set analysis</strong></p>
<p>This option provides a possibility to compare plots for different samples.
Supply a tab separated text file, with the following structure:</p>
<blockquote>
<div><p>Columns: gene_attribute (e.g. Name) -&gt; geneset</p>
<p>Rows: value_of_the_attribute -&gt; geneset_name</p>
</div></blockquote>
<p>Note, the gene_attribute is the attribute name in the gtf or gff file.
In case of gff, the attribute in the cds feature will be considered.
With this option, fivepseq will generate a separate plotting directory called genesets, with tabbed-plots to either compare samples for each geneset, or genesets for each sample. The counts folder will also be divided according to the geneset used. The default folder will be named protein_coding.</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>-gs/-geneset
    default: None
</pre></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">fivepseq</a></h1>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="preparing_data.html">Preparing Data</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Fivepseq usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#command-line-arguments">Command line arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#optional-arguments">Optional arguments</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="interpreting_output.html">Output interpretation</a></li>
<li class="toctree-l1"><a class="reference internal" href="advanced.html">Advanced</a></li>
<li class="toctree-l1"><a class="reference internal" href="citing.html">Citing</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="preparing_data.html" title="previous chapter">Preparing Data</a></li>
      <li>Next: <a href="interpreting_output.html" title="next chapter">Output interpretation</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2019, Lilit Nersisyan.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.5</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/usage_fivepseq.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>